tests/components/zha/test_channels.py

```diff
@@ -159,4 +171,203 @@ def test_channel_registry():
+        assert ch_3 in available
+
+        ep_channels.claimed_channels[ch_1.id] = ch_1
+        available = ep_channels.unclaimed_channels()
+        assert ch_1 not in available
+        assert ch_2 not in available
+        assert ch_3 in available
+
+        ep_channels.claimed_channels[ch_3.id] = ch_3
+        available = ep_channels.unclaimed_channels()
+        assert ch_1 not in available
+        assert ch_2 not in available
+        assert ch_3 not in available
+
+
+def test_epch_claim_channels(channel):
+    """Test channel claiming."""
+
+    ch_1 = channel(zha_const.CHANNEL_ON_OFF, 6)
+    ch_2 = channel(zha_const.CHANNEL_LEVEL, 8)
+    ch_3 = channel(zha_const.CHANNEL_COLOR, 768)
+
+    ep_channels = zha_channels.ChannelPool(
+        mock.MagicMock(spec_set=zha_channels.Channels), mock.sentinel.ep
+    )
+    all_channels = {ch_1.id: ch_1, ch_2.id: ch_2, ch_3.id: ch_3}
+    with mock.patch.dict(ep_channels.all_channels, all_channels, clear=True):
+        assert ch_1.id not in ep_channels.claimed_channels
+        assert ch_2.id not in ep_channels.claimed_channels
+        assert ch_3.id not in ep_channels.claimed_channels
+
+        ep_channels.claim_channels([ch_2])
+        assert ch_1.id not in ep_channels.claimed_channels
+        assert ch_2.id in ep_channels.claimed_channels
+        assert ep_channels.claimed_channels[ch_2.id] is ch_2
+        assert ch_3.id not in ep_channels.claimed_channels
+
+        ep_channels.claim_channels([ch_3, ch_1])
+        assert ch_1.id in ep_channels.claimed_channels
+        assert ep_channels.claimed_channels[ch_1.id] is ch_1
+        assert ch_2.id in ep_channels.claimed_channels
+        assert ep_channels.claimed_channels[ch_2.id] is ch_2
+        assert ch_3.id in ep_channels.claimed_channels
+        assert ep_channels.claimed_channels[ch_3.id] is ch_3
+        assert "1:0x0300" in ep_channels.claimed_channels
+
+
+@mock.patch("homeassistant.components.zha.core.channels.ChannelPool.add_relay_channels")
+@mock.patch(
+    "homeassistant.components.zha.core.discovery.PROBE.discover_entities",
+    mock.MagicMock(),
+)
+def test_ep_channels_all_channels(m1, zha_device_mock):
+    """Test EndpointChannels adding all channels."""
+    zha_device = zha_device_mock(
+        {
+            1: {"in_clusters": [0, 1, 6, 8], "out_clusters": [], "device_type": 0x0000},
+            2: {
+                "in_clusters": [0, 1, 6, 8, 768],
+                "out_clusters": [],
+                "device_type": 0x0000,
+            },
+        }
+    )
+    channels = zha_channels.Channels(zha_device)
+
+    ep_channels = zha_channels.ChannelPool.new(channels, 1)
+    assert "1:0x0000" in ep_channels.all_channels
+    assert "1:0x0001" in ep_channels.all_channels
+    assert "1:0x0006" in ep_channels.all_channels
+    assert "1:0x0008" in ep_channels.all_channels
+    assert "1:0x0300" not in ep_channels.all_channels
+    assert "2:0x0000" not in ep_channels.all_channels
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5051600)